pybamm/solvers/processed_casadi_variable.py

```diff
@@ -0,0 +1,223 @@
+#
+# Processed Variable class
+#
+import casadi
+import numbers
+import numpy as np
+
+
+class ProcessedCasadiVariable(object):
```

Comment:
It is feasible that we could implement this (a symbolic processed variable) for non-casadi solvers, so I don't think we should use Casadi in the class name. Perhaps `ProcessedSymbolicVariable`?